warehouse/accounts/forms.py

```diff
@@ -14,6 +14,8 @@
 import wtforms
 import wtforms.fields.html5
 
+from pyramid.threadlocal import get_current_request
```

Comment:
not optimal but I'm not sure how else to get the link in front of the user in the validation error

Reply:
Could pass the ``request`` into the form like we do the ``breach_service``, or pass in the link directly.

Alternatively you could add a method to the breach service that returns a failure message, and the factory for the breach service could then be modified to accept a request (or again, the url directly).

I kind of like the idea of putting the failure message method (or even just a property. Zope interface supports properties too:

```python
from zope.interface import Attribute, Interface

class IFoo(Interface):
    failure_message = Attribute("The failure message")


@implementer(IFoo)
class Foo:

    def __init__(self, url):
        self._url = url

    @property
    def failure_message(self):
        return f"Bad things happened, see: {self._url}"
```

Reply:
Nice thing about making the failure message part of the service, is it will let us tailor the message to different services, if we end up using a different API than HIBP or something.

Reply:
Although if it's completely generic like it is now, maybe it would be better on the form? I dunno.